api/ccapi/include/layer.h

```diff
@@ -97,8 +98,14 @@ class Layer {
    *            23. gamma_initializer : string (type),
    *            24. beta_initializer" : string (type)
    *            25. modelfile : model file for loading config for backbone layer
-   *            26. input_layers" : string (type)
-   *            27. output_layers" : string (type)
+   *            26. input_layers : string (type)
+   *            27. output_layers : string (type)
+   *            28. trainable :
+   *            29. flip_direction
+   *            30. random_translate
+   *            31. in_dim : int ( input dimension for embedding layer )
```

Comment:
i wonder how `in_dim` is different from `input_shape`?

Reply:
yes. it differs. in_dim is vocab size. and the input size is related with input length. and the out dim is the size of embedding space.

Reply:
Then how about rename in_dim to vocab_size or add in comments?

> the out dim is the size of embedding space.

Is out_dim supposed to match with output_dimension of the layer? Please add such invariants in the comments of this property.

Reply:
as you can see, those number is just unsigned integer. not the dim. output dimension must be composed with batch, in_length, out_dim.

Reply:
ok.